explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-  const badgeCount = messages
-    .filter(m => m.unread)
-    .length;
-  Badge.set(badgeCount);
-}
+### Nested scopes
 
-// Adds a new message and updates the badge count.
-function addMessage(message) {
-  messages.push(message);
-  updateBadgeCount();
-}
+Since we allow badges to be scoped to different, potentially nested, URLs, it
+means that a particular page can be subject to more than one badge at a time.
+In this case, the user agent should display only the badge with the most
+specific URL.
 
-// Create a new socket, connected to our messages endpoint.
-const socket = new WebSocket('ws://example.com/messages');
-socket.onmessage = (event) => {
-  // Parse the message and add it to our list.
-  const chatMessage = JSON.parse(event.data);
-  addMessage(chatMessage);
-};
-```
+Therefore, clearing a badge (either by calling `Badge.clear()` or
+`Badge.set(0)`) does *not necessarily* mean that no badge will be displayed; by
+erasing a badge at one level, a page may inherit a badge from a higher level.
 
-##### Setting a separate badge for the app and a specific page (as in the case of GitHub notifications and PR statuses).
-On all pages of this site, we wish to display the notification count, except for `/status/{number}`, where we should instead display `flag` if the page's status is `ready` or nothing, if it is not.
-
-The main page of our site https://example.com/
-```html
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-    <title>Main Page</title>
-  </head>
-  <body>
-    <script>
-      // There are ten unread notifications.
-      // This was injected by the server.
-      Badge.set(10);
-    </script>
-  </body>
-</html>
-```
+For example, consider a site that has made the following two calls:
 
-On https://example.com/status/1
-```html
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-    <title>Main Page</title>
-  </head>
-  <body>
-    <script>
-      // `/status/39/` => `39`
-      const statusNumber = location.pathname
-        .split('/') // Get path segments
-        .filter(s => s) // which are non-empty
-        .pop(); // and take the last one.
-
-      const badgeOptions = {
-        scope: `/status/${statusNumber}`
-      };
-
-      const statusInfo = {
-        status: 'ready',
-        author: 'me',
-        url: `https://example.com/status/${statusNumber}`
-      };
-
-      if (statusInfo.status === 'ready')
-        Badge.set(badgeOptions);
-      else
-        Badge.clear(badgeOptions);
-
-    </script>
-  </body>
-</html>
-```
+* `Badge.set(6, {scope: '/users/'});`
+* `Badge.set(2, {scope: '/users/1'});`
+
+Now all pages in '/users/' show the badge "6", except for `/users/1`, which
+shows the badge "2".
+
+Now if we see `Badge.clear({scope: '/users/1'})`, the pages under `/users/1`
+will start showing the badge "6" since that badge is still in effect. If instead
+we see `Badge.clear({scope: '/users/'})`, the pages under `/users/1` will still
+show the badge "2", *even if `clear` is called from one of those pages*.
```

Comment:
Possibly link to the relevant example /docs/examples.md#badging-for-multiple-apps-on-the-same-origin-as-in-the-case-of-multiple-github-pages-pwas?

Reply:
Done.